src/WingetCreateCore/Common/PackageParser.cs

```diff
@@ -255,16 +340,47 @@ public static bool UpdateInstallerNodes(InstallerManifest installerManifest, IEn
             }
         }
 
-        private static bool ParsePackageAndGenerateInstallerNodes(string path, string url, List<Installer> installers, Manifests manifests)
+        private static bool ParsePackageAndGenerateInstallerNodes(
+            string path,
+            string url,
+            List<Installer> installers,
+            Manifests manifests,
+            ref List<DetectedArch> detectedArchOfInstallers)
         {
             var installer = new Installer();
             installer.InstallerUrl = url;
             installer.InstallerSha256 = GetFileHash(path);
             installer.Architecture = GetMachineType(path)?.ToString().ToEnumOrDefault<InstallerArchitecture>() ?? InstallerArchitecture.Neutral;
 
-            return ParseExeInstallerType(path, installer, installers) ||
-                ParseMsix(path, installer, manifests, installers) ||
+            bool parseMsixResult = false;
+
+            bool parseResult = ParseExeInstallerType(path, installer, installers) ||
+                (parseMsixResult = ParseMsix(path, installer, manifests, installers)) ||
                 ParseMsi(path, installer, manifests, installers);
+
+            if (parseMsixResult)
+            {
+                // Skip architecture override for msix installers as there can be more than one installer in a bundle
+                var nonMsixInstallerUrls = detectedArchOfInstallers.Select(i => i.Url).ToList();
+                var msixInstallers = installers.Where(i => !nonMsixInstallerUrls.Contains(i.InstallerUrl));
+                foreach (var msixInstaller in msixInstallers)
+                {
+                    var msixArch = msixInstaller.Architecture;
+                    detectedArchOfInstallers.Add(new DetectedArch(msixInstaller.InstallerUrl, msixArch, msixArch));
```

Comment:
I don't think you need to add/log these at all? Since we're not overriding these, we don't need to worry about warning the user about them, right?

Reply:
I know this is slightly confusing but I need to add these to detectedArchOfInstallers or else matching will fail on PackageParser Line 205 - 215. This is because ParseMsix adds new installers from the bundle, which is not picked up by detectedArchOfInstallers so when it tries to match by installerURL and Arch, it fails. since those installers aren't present. 

These won't get shown in the logs because urlArch == binaryArch so logging isn't affected.

Reply:
detectedArchOfInstallers.AddRange(
    installers
    .Where(i => i.InstallerUrl == url)
    .Select(i => new DetectedArch(i.InstallerUrl, i.Architecture, i.Architecture))
)

Reply:
Test

Reply:
Changed to suggestion